### PR TITLE
Add release-branch-forward script

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind design

#### What this PR does / why we need it:
Add a first golang based script for release branch automation.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
Maybe we can try the tool manually in the first place to see if it fits our requirements. 
/cc @haircommander @mrunalp 
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: scripts/release-branch-forward/release-branch-forward.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+)
+
+var dryRun bool
+
+func init() {
+	flag.BoolVar(
+		&dryRun,
+		"dry-run",
+		true,
+		"do not relly push, just do a dry run",
```
Comment (TomSweeneyRedHat) [inline scripts/release-branch-forward/release-branch-forward.go]: nit relly->really

File: scripts/release-branch-forward/release-branch-forward.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		return nil
+	}
+
+	// Checkout the release branch
+	repo, err := kgit.OpenRepo(".")
+	if err != nil {
+		return errors.Wrap(err, "unable to open this repository")
+	}
+	if dryRun {
+		logrus.Info("Setting repository to do only a dry-run")
```
Comment (TomSweeneyRedHat) [inline scripts/release-branch-forward/release-branch-forward.go]: if you fix the other nit, "do only" -> "only do"

**Comment by TomSweeneyRedHat** [comment]
couple of nits, otherwise LGTM

File: scripts/release-branch-forward/release-branch-forward.go (hunk 1 above)
Comment (saschagrunert) [reply scripts/release-branch-forward/release-branch-forward.go]: Fixed :+1:

File: scripts/release-branch-forward/release-branch-forward.go (hunk 2 above)
Comment (saschagrunert) [reply scripts/release-branch-forward/release-branch-forward.go]: Fixed as well :+1:

**Comment by dougsland** [comment]
Can we add a README file or/and --help option  how to test/use the tool?

**Comment by rhatdan** [comment]
LGTM

**Comment by saschagrunert** [comment]
> Can we add a README file or/and --help option how to test/use the tool?

Before adding documentation and further automation I'd like to have maybe @haircommander do a test with the current 1.18 release branch. As a follow-up we can add a CircleCI job which runs periodically and updates the release branch every day.

File: scripts/release-branch-forward/release-branch-forward.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/release/pkg/command"
+	kgit "k8s.io/release/pkg/git"
+)
+
+const (
+	remote              = "https://github.com/cri-o/cri-o"
+	git                 = "git"
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply scripts/release-branch-forward/release-branch-forward.go]: Yes, I added a check for the binaries. But I think having them in `$PATH` is enough. :)

**Comment by rhatdan** [comment]
@dougsland @haircommander @umohnani8 @mrunalp This one ready to go in?

**Comment by saschagrunert** [comment]
Rebased to resolve the conflicts

**Comment by saschagrunert** [comment]
@haircommander do you want to give the script a try on the latest release-1.18 branch? (I think we have to sort-out open PRs afterwards) If everything works as intended I would setup the automation as well.

**Comment by haircommander** [comment]
we need to wait at least for https://github.com/cri-o/cri-o/pull/3399, otherwise the merge conflicts are too much to automate

**Comment by haircommander** [comment]
it may be the case we need to wait until we cut final 1.18, and branch for 1.19 to set up the automation

**Pull request merged**